pkg/webui/console/store/middleware/logics/events.js

```diff
@@ -68,31 +67,42 @@ const createEventsConnectLogics = function (
         const connecting = status === CONNECTION_STATUS.CONNECTING
         if (connected || connecting) {
           reject()
+          return
         }
 
         allow(action)
       },
-      async process ({ action }, dispatch, done) {
+      async process ({ getState, action }, dispatch, done) {
         const { id } = action
 
         try {
           channel = await onEventsStart([ id ])
+
           channel.on('start', () => dispatch(startEventsSuccess(id)))
           channel.on('event', message => dispatch(getEventSuccess(id, message)))
-          channel.on('error', error => dispatch(getEventFailure(id, error)))
+          channel.on('error', function (error) {
+            const status = selectEntityEventsStatus(getState(), id)
+            if (status === CONNECTION_STATUS.CONNECTING) {
+              dispatch(startEventsFailure(id, error))
+              done()
+            } else {
+              dispatch(getEventFailure(id, error))
```

Comment:
if we stick to this implementation, should `done()` be called here as well?

Reply:
Not sure about this. The way I see it, an error during retrieval of an event does not necessarily mean that the stream has closed altogether.

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/pull/974/files#diff-1ef843661c3cb6cc93d5f07c76d59dd8R97

Reply:
Indeed. It should call `done()` then.